### ISSUE TITLE
Enhancement: [@typescript-eslint/no-explicit-any] add ignoreGenericParameters

### ISSUE THREAD

**Issue by splincode** [typescript-eslint/typescript-eslint#11515]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [x] I have searched the [current rule list](https://typescript-eslint.io/rules/#rules) and found no rules that match my proposal.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [x] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Link to the rule's documentation

https://typescript-eslint.io/rules/no-explicit-any/

### Description

I have a lot of generics 

### Fail

```TypeScript
const a: any = fn('');

function fn(b: any) {
}
```

### Pass

```TypeScript
export class TuiTableHead<T extends Partial<Record<keyof T, any>>> {
  public readonly table!: TuiTableDirective<any>;
}
```

### Additional Info

```ts
const defaultOptions: Options = [
  { fixToUnknown: false, ignoreRestArgs: false, ignoreGenericParameters: false },
];
```

**Comment by JoshuaKGoldberg** [comment]
> I have a lot of generics

You and me both. 😜 

I suspect the referenced code structure is actually covered by #11063 -> #11199 and the linked issues. Specifically, I think this issue is a duplicate of #642 and #7751. Do you have anything to add on why the strategies suggested in those issues wouldn't work here?

**Comment by splincode** [comment]
Oh, thank you
